lib/logstash/outputs/s3.rb

```diff
@@ -84,6 +86,9 @@ class LogStash::Outputs::S3 < LogStash::Outputs::Base
   # S3 bucket
   config :bucket, :validate => :string
 
+  # endpoint
+  config :endpoint, :validate => :string
```

Comment:
It seems like you'd want to reject a config where both endpoint and region are specified. Right? Can you add that?

Also, can you add more detailed docs for this new setting? For users only familiar with AWS, I imagine this setting may be confusing. Further, describing the format (a url) would be good as well.